wdl-engine/tests/tasks.rs

```diff
@@ -421,21 +415,40 @@ fn compare_evaluation_results(
     Ok(())
 }
 
-/// Creates a string from the given diagnostic.
-fn diagnostic_to_string(document: &Document, path: &str, diagnostic: &Diagnostic) -> String {
-    let source = document.root().text().to_string();
-    let file = SimpleFile::new(path, &source);
-
-    let mut buffer = Buffer::no_color();
-    term::emit(
-        &mut buffer,
-        &Config::default(),
-        &file,
-        &diagnostic.to_codespan(),
-    )
-    .expect("should emit");
-
-    String::from_utf8(buffer.into_inner()).expect("should be UTF-8")
+/// Creates a string from the given evaluation error.
+fn eval_error_to_string(e: &EvaluationError) -> String {
+    match e {
+        EvaluationError::Source(e) => {
+            let mut files = SimpleFiles::new();
```

Comment:
Discussion point, but not necessarily for this PR: should we be keeping a `SimpleFiles` associated with the analysis itself rather than creating it on demand?

Reply:
There's no dependency on `codespan_reporting` from `wdl-analysis` and I don't think it's necessary to introduce one here; given that we only construct a `SimpleFiles` once upon evaluation error and then immediately exit.

Generally the diagnostic representation from `wdl-grammar` and down into `wdl-engine` has been agnostic with regards to _how_ a diagnostic will be displayed to the user. I'd prefer the `codespan_reporting` dependency to be in the executable projects and not the libs (with the exception of a helper in `wdl-grammar` for converting to a `codespan_reporting::Diagnostic`, which is optional and enabled from the executables).

Reply:
Sure that makes sense.